src/test/java/org/opentripplanner/routing/core/RoutingContext_Destroy_Test.java

```diff
@@ -0,0 +1,147 @@
+package org.opentripplanner.routing.core;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+import org.junit.Before;
+import org.junit.Test;
+import org.opentripplanner.common.geometry.GeometryUtils;
+import org.opentripplanner.common.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.common.model.GenericLocation;
+import org.opentripplanner.routing.edgetype.StreetEdge;
+import org.opentripplanner.routing.edgetype.StreetTraversalPermission;
+import org.opentripplanner.routing.edgetype.TemporaryEdge;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.impl.DefaultStreetVertexIndexFactory;
+import org.opentripplanner.routing.vertextype.IntersectionVertex;
+import org.opentripplanner.routing.vertextype.StreetVertex;
+import org.opentripplanner.routing.vertextype.TemporaryVertex;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class RoutingContext_Destroy_Test {
+    private final GeometryFactory gf = GeometryUtils.getGeometryFactory();
+    private RoutingRequest request;
+    private RoutingContext subject;
+
+    // Given:
+    // - a graph with 3 intersections/vertexes
+    private final Graph g = new Graph();
+
+    private final StreetVertex a = new IntersectionVertex(g, "A", 1.0, 1.0);
+
+    private final StreetVertex b = new IntersectionVertex(g, "B", 0.0, 1.0);
+
+    private final StreetVertex c = new IntersectionVertex(g, "C", 1.0, 0.0);
+
+    private final List<Vertex> permanentVertexes = Arrays.asList(a, b, c);
+
+    // - And travel *origin* is 0,4 degrees on the road from B to A
+    private final GenericLocation from = new GenericLocation(1.0, 0.4);
+
+    // - and *destination* is slightly off 0.7 degrees on road from C to A
+    private final GenericLocation to = new GenericLocation(0.701, 1.001);
+
+    // - and some roads
+    @Before public void setup() {
+        createStreetEdge(a, b, "a -> b");
+        createStreetEdge(b, a, "b -> a");
+        createStreetEdge(a, c, "a -> c");
+        g.index(new DefaultStreetVertexIndexFactory());
+    }
+
+    @Test public void temporary_changes_is_removed_from_graph_on_context_destroy() {
```

Comment:
Code style in the rest of the project is camelCase.

Reply:
Indeed this is even required by the project code style: https://opentripplanner.readthedocs.io/en/latest/Developers-Guide/#code-style